pkg/driver/bigquery/bigquery_test.go

```diff
@@ -0,0 +1,379 @@
+	require.NoError(t, err)
+
+	count := 0
+	err = db.QueryRow("select count(*) from test_migrations").Scan(&count)
+	require.NoError(t, err)
+	require.Equal(t, 0, count)
+
+	// insert migration
+	err = drv.InsertMigration(db, "abc1")
+	require.NoError(t, err)
+
+	err = db.QueryRow("select count(*) from test_migrations where version = 'abc1'").
+		Scan(&count)
+	require.NoError(t, err)
+	require.Equal(t, 1, count)
+}
+
+func TestBigQueryDeleteMigration(t *testing.T) {
+	drv := testBigQueryDriver(t)
+	drv.migrationsTableName = "test_migrations"
+
+	db := prepTestBigQueryDB(t)
+	defer dbutil.MustClose(db)
+
+	err := drv.CreateMigrationsTable(db)
+	require.NoError(t, err)
+
+	_, err = db.Exec(`insert into test_migrations (version)
+		values ('abc1'), ('abc2')`)
+	require.NoError(t, err)
+
+	err = drv.DeleteMigration(db, "abc2")
+	require.NoError(t, err)
+
+	count := 0
+	err = db.QueryRow("select count(*) from test_migrations").Scan(&count)
+	require.NoError(t, err)
+	require.Equal(t, 1, count)
+}
+
+func TestBigQueryPingError(t *testing.T) {
+	drv := testBigQueryDriver(t)
+
+	// drop any existing database
+	err := drv.DropDatabase()
+	require.NoError(t, err)
+
+	// ping database
+	err = drv.Ping()
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "dataset dbmate_test is not found")
+}
+
+func TestBigQueryPingSuccess(t *testing.T) {
+	drv := testBigQueryDriver(t)
+
+	db := prepTestBigQueryDB(t)
+	defer dbutil.MustClose(db)
+
+	// ping database
+	err := drv.Ping()
+	require.NoError(t, err)
+}
+
+func TestBigQueryMigrationsTableExists(t *testing.T) {
+	drv := testBigQueryDriver(t)
+	drv.migrationsTableName = "test_migrations"
+
+	db := prepTestBigQueryDB(t)
+	defer dbutil.MustClose(db)
+
+	exists, err := drv.MigrationsTableExists(db)
+	require.NoError(t, err)
+	require.Equal(t, false, exists)
+
+	err = drv.CreateMigrationsTable(db)
+	require.NoError(t, err)
+
+	exists, err = drv.MigrationsTableExists(db)
+	require.NoError(t, err)
+	require.Equal(t, true, exists)
+}
+
+func TestGoogleBigQueryDumpSchema(t *testing.T) {
+	t.Run("default migrations table", func(t *testing.T) {
+		drv := testGoogleBigQueryDriver(t)
+
+		// prepare database
+		db := prepTestGoogleBigQueryDB(t)
+		defer dbutil.MustClose(db)
+		err := drv.CreateMigrationsTable(db)
+		require.NoError(t, err)
+
+		// insert migration
+		err = drv.InsertMigration(db, "abc1")
+		require.NoError(t, err)
+		err = drv.InsertMigration(db, "abc2")
+		require.NoError(t, err)
+
+		// DumpSchema should return schema
+		config, err := drv.getConfig(db)
+		require.NoError(t, err)
+
+		schema, err := drv.DumpSchema(db)
+		require.NoError(t, err)
+		require.Contains(t, string(schema), fmt.Sprintf("CREATE TABLE `%s.%s.schema_migrations`", config.projectID, config.dataSet))
+		require.Contains(t, string(schema), "\n--\n"+
+			"-- Dbmate schema migrations\n"+
+			"--\n\n"+
+			"INSERT INTO schema_migrations (version) VALUES\n"+
+			"    ('abc1'),\n"+
+			"    ('abc2');\n")
+	})
+}
```

Comment:
@mii9000 Try this test out and let me know what you think.

Authenticate with a service account that has the necessary IAM roles (see this [README](...) for details) and put a copy of the credentials JSON in `fixtures/bigquery/credentials.json` and give this a try:

```
$ make test \
  GOOGLE_APPLICATION_CREDENTIALS=/src/fixtures/bigquery/credentials.json \
 
 GOOGLE_BIGQUERY_TEST_URL=bigquery://$PROJECT_ID/$LOCATION/$DATASET
```

Replace `$PROJECT_ID/$LOCATION/$DATASET` with the project ID, region/location, and a test dataset name (remember, the dbmate tests will drop and create the dataset, so don't run tests against a dataset with data you can't afford to lose).

Reply:
![image](https://github.com/amacneil/dbmate/assets/6401874/bd2a1e58-c572-4316-ab2b-dc432aba7c86)
@dossy All tests passed! Would it be better if the tests against the actual big query was in a separate file?

Reply:
So what else is required to get this PR merged @dossy?

Reply:
@mii9000 Our test coverage against the actual BigQuery is pretty light, so I'm not especially confident that dbmate will work as expected through actual user interactions.  Users do all kinds of things that we don't have automated tests for.

It would be helpful if anyone who plans to actually use dbmate with BigQuery could pick up a dev build of it and actually start using it, and give feedback before it's merged -- there's no clear way for us to indicate that a particular driver may not be production-ready, yet.

I'm most concerned of dbmate dropping a user's entire dataset in production, when they didn't expect it to happen, or something terribly unfortunate like that.

Reply:
But the addition of bq driver is same as others. Other drivers are also dropping a database. Why bq should be any different? If its a concern then remove the drop database functionality?

Reply:
All the tests are working against the emulator. The only one missing is the dump schema one which can be covered with unit testing at least without requiring actual bq

Reply:
@dossy I can share a credentials.json from a trial account I created with you that will help you run the tests as well. Let me know if that will suffice with enough testing.

Reply:
@mii9000 I have my own service account JSON that I'm testing with, but I have no actual projects on BigQuery that would use dbmate. I'd like someone who intends to use this for real projects to give it a try and give feedback on whether it works for them or not.

Reply:
I contributed this PR because my team intends to use this in Production. But if we are waiting for people to actually use this in Production before merging this PR or any PR going forward then at that rate it will take forever to run PRs.

I don't think it's completely up to maintainers or owners to validate how this tool will be used end-to-end. This PR checked all the boxes i.e. automated and manual tests, code reviews, standards, etc. I don't find this as a valid reason to delay releasing a new version based on this PR quite frankly. @dossy @amacneil

Reply:
@mii9000 Do you require a tagged release before your team can use this?  Can you start with a build from this PR branch, and exercise the functionality with real migrations, and see if there's anything that doesn't work as expected, before this PR gets merged?

That's all I was hoping for, one last sanity check with real migrations before merging this.

If @amacneil is comfortable merging this as it currently is, that's up to him.  I'm a little more cautious, that's just my personal preference.